pages/generatedFilesQuestion/generatedFilesQuestion.sql

```diff
@@ -0,0 +1,13 @@
+-- BLOCK select_variant
+SELECT
+    v.*
+FROM
+    variants AS v
+WHERE
+    v.id = $variant_id
+    -- We check both `question_id` and `instance_question_id` to make sure that
+    -- this variant is actually accessible by the user. Both will have been
+    -- validated by middleware. For the instructor preview page, we don't have
+    -- an instance question, so we ignore it in that case.
+    AND v.question_id = $question_id
+    AND (($has_instance_question IS FALSE) OR (v.instance_question_id = $instance_question_id));
```

Comment:
I couldn't find a better way to write this, since unfortunately it seems like we can't use query parameters multiple times in the same query. I didn't want to make this a full-blown sproc, so this is where I ended up. I'm tired and kinda sick though, so I'm probably missing some obviously-better way to write this 😅

Reply:
I might be misunderstanding what you mean here, but you _can_ use `$instance_question_id` multiple times in the same query, for example. The value of `xxx` just gets templated in for every occurrence of `$xxx`.